### PR TITLE
feat: make @sapphire/stopwatch browser compitible

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
fixes: #271

### CODE REVIEW

File: packages/stopwatch/src/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
 	"extends": "../../ts-config/extra-strict-without-decorators.json",
 	"compilerOptions": {
+		"lib": ["dom", "dom.iterable", "esnext"],
```
Comment (favna) [inline packages/stopwatch/src/tsconfig.json:4]:
```suggestion
		"lib": ["dom", "esnext"],
```

Doesn't `dom` also include `dom.iterable`?

File: packages/stopwatch/tsup.config.ts [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+					js: "import { performance } from 'node:perf_hooks';"
+				};
+				break;
+			}
+			case 'iife': {
+				// iife doesn't need any fetch replacements
+				break;
+			}
+		}
+	}
```
Comment (favna) [inline packages/stopwatch/tsup.config.ts:25]: CJS and ESM don't need imports either, `performance` is available globally in Node:

image

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — CHANGES_REQUESTED** [review]
Also add to the top of `index.ts`:
```ts
// eslint-disable-next-line spaced-comment
/// <reference lib="dom" />
```

the added `dom` in tsconfig doesn't work for end-users

**Review by kyranet — DISMISSED** [review]

**Review by vladfrangu — DISMISSED** [review]

File: packages/stopwatch/tsup.config.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
 import { createTsupConfig } from '../../scripts/tsup.config';
 
-export default createTsupConfig({ format: ['esm', 'cjs'] });
+export default createTsupConfig({
+	globalName: 'Stopwatch'
```
Comment (favna) [inline packages/stopwatch/tsup.config.ts:4]:
```suggestion
	globalName: 'SapphireStopwatch'
```

`Sapphire` prefix to retain consistency

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**